### PR TITLE
Add test to check data and index types compatibility

### PR DESCRIPTION
**Pull request by bziobrowski**
PR adds integration test (disabled by default) that checks compatibility between combinations of:
- data type - INT, LONG, FLOAT, DOUBLE, BIG_DECIMAL, BOOLEAN, TIMESTAMP, STRING, JSON, BYTES, MAP (all variants)
- cardinality - single value, multiple values
- encoding - raw or dictionary
- index types - timestamp, bloom, fst, geo, inverted, json, native_text, text, range, vector
and reports the outcome .

### CODE REVIEW

**Comment by npawar** [comment]
Why disable by default? this would be a good thing to have running to catch combination regressions?

**Comment by Jackie-Jiang** [comment]
Nice! This can give us big confidence!
How long does it take to run the test? Both in your local environment and in Github Actions?

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TableIndexingTest.java [hunk 1]
```diff
@@ -0,0 +1,776 @@
...
+            "JSON index can only be applied to single value column of STRING or JSON data type!");
+      }
+
+      if ("vector".equals(indexType) && (field.getDataType() != DataType.FLOAT || field.isSingleValueField())) {
+        throw new RuntimeException("VECTOR index can only be applied to Float Array columns");
+      }
+
+      if (("text".equals(indexType) || "native_text".equals(indexType)) && field.getDataType() != DataType.STRING) {
+        throw new RuntimeException("Text index is currently only supported on STRING columns");
+      }
```
Comment (gortiz) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TableIndexingTest.java:203]: I don't understand this. What does this failure means? If we want to skip these cases we can either:
1. Preferred option: Filter them out in `fieldsAndIndexTypes`
2. Alternatively: We can throw SkipException, which makes TestNG to skip the test.

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TableIndexingTest.java [hunk 2]
```diff
@@ -0,0 +1,776 @@
...
+                  "indexes": {
+                    "bloom": {}
+                  }
+                },
+                ...
+              ],
+              ...
+            } */
+          // no params
+          indexes.put("bloom", new ObjectNode(JsonNodeFactory.instance));
```
Comment (gortiz) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TableIndexingTest.java:236]:
```suggestion
          indexes.put("bloom", JsonUtils.newObjectNode());
```

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TableIndexingTest.java [hunk 3]
```diff
@@ -0,0 +1,776 @@
...
+                  }
+                }
+              ],
+              ...
+            }
+             */
+          ObjectNode resolutions = new ObjectNode(JsonNodeFactory.instance);
+          ArrayNode res = new ArrayNode(JsonNodeFactory.instance);
+          res.add(13).add(5).add(6);
+          resolutions.put("resolutions", res);
```
Comment (gortiz) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TableIndexingTest.java:271]:
```suggestion
          ObjectNode resolutions = JsonUtils.stringToJsonNode("{\"resolutions\": [13, 5, 6]}");
```

File: pinot-core/src/test/java/org/apache/pinot/core/data/manager/TableIndexingTest.java [hunk 4]
```diff
@@ -0,0 +1,717 @@
...
+    createTestCases();
+    readExpectedResults();
+  }
+
+  private void createTestCases() {
+    String[] indexTypes =
+        {
+            "timestamp_index", "bloom_filter", "fst_index", "h3_index", "inverted_index", "json_index",
+            "native_text_index", "text_index", "range_index", "startree_index", "vector_index"
+        };
```
Comment (Jackie-Jiang) [inline pinot-core/src/test/java/org/apache/pinot/core/data/manager/TableIndexingTest.java:107]: (nit) We usually put `{` in the previous line
```suggestion
    String[] indexTypes = {
        "timestamp_index", "bloom_filter", "fst_index", "h3_index", "inverted_index", "json_index",
        "native_text_index", "text_index", "range_index", "startree_index", "vector_index"
    };
```

File: pinot-core/src/test/java/org/apache/pinot/core/data/manager/TableIndexingTest.java [hunk 5]
```diff
@@ -0,0 +1,717 @@
...
+    }
+  }
+
+  private void readExpectedResults()
+      throws IOException {
+    List<String> expected = readExpectedFromFile();
+    // parse csv lines, e.g. INT;sv;raw;timestamp_index;true;
+    for (int i = 1; i < expected.size(); i++) {
+      String line = expected.get(i);
+      if (line.length() == 0) {
```
Comment (Jackie-Jiang) [inline pinot-core/src/test/java/org/apache/pinot/core/data/manager/TableIndexingTest.java:127]: (nit) My IDE usually gives warning on this
```suggestion
      if (line.isEmpty()) {
```

**Review by Jackie-Jiang — APPROVED** [review]
Nice!

**Pull request merged**